atomics/include/desul/atomics/Fetch_Op_SYCL.hpp

```diff
@@ -11,52 +11,125 @@ SPDX-License-Identifier: (BSD-3-Clause)
+    return dest_ref++;                                                       \
+  }                                                                          \
+  template <class MemoryOrder>                                               \
+  TYPE device_atomic_inc_fetch(TYPE* dest, MemoryOrder, MemoryScopeDevice) { \
+    sycl_atomic_ref<TYPE, MemoryOrder, MemoryScopeDevice> dest_ref(*dest);   \
+    return ++dest_ref;                                                       \
+  }                                                                          \
+  template <class MemoryOrder>                                               \
+  TYPE device_atomic_inc_fetch(TYPE* dest, MemoryOrder, MemoryScopeCore)   { \
+    sycl_atomic_ref<TYPE, MemoryOrder, MemoryScopeCore> dest_ref(*dest);     \
+    return ++dest_ref;                                                       \
+  }
+
+#define DESUL_IMPL_SYCL_ATOMIC_DEC(TYPE)                                     \
+  template <class MemoryOrder>                                               \
+  TYPE device_atomic_fetch_dec(TYPE* dest, MemoryOrder, MemoryScopeDevice) { \
+    sycl_atomic_ref<TYPE, MemoryOrder, MemoryScopeDevice> dest_ref(*dest);   \
+    return dest_ref--;                                                       \
+  }                                                                          \
+  template <class MemoryOrder>                                               \
+  TYPE device_atomic_fetch_dec(TYPE* dest, MemoryOrder, MemoryScopeCore  ) { \
+    sycl_atomic_ref<TYPE, MemoryOrder, MemoryScopeCore> dest_ref(*dest);     \
+    return dest_ref--;                                                       \
+  }                                                                          \
+  template <class MemoryOrder>                                               \
+  TYPE device_atomic_dec_fetch(TYPE* dest, MemoryOrder, MemoryScopeDevice) { \
+    sycl_atomic_ref<TYPE, MemoryOrder, MemoryScopeDevice> dest_ref(*dest);   \
+    return --dest_ref;                                                       \
+  }                                                                          \
+  template <class MemoryOrder>                                               \
+  TYPE device_atomic_dec_fetch(TYPE* dest, MemoryOrder, MemoryScopeCore)   { \
+    sycl_atomic_ref<TYPE, MemoryOrder, MemoryScopeCore> dest_ref(*dest);     \
+    return --dest_ref;                                                       \
+  }
+
+#define DESUL_IMPL_SYCL_ATOMIC_INC_DEC(TYPE) \
+  DESUL_IMPL_SYCL_ATOMIC_INC(TYPE)           \
+  DESUL_IMPL_SYCL_ATOMIC_DEC(TYPE)
+// clang-format on
+
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(int)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(unsigned int)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(long)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(unsigned long)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(long long)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(unsigned long long)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(float)
+DESUL_IMPL_SYCL_ATOMIC_INC_DEC(double)
```

Comment:
I expect that means we had no coverage for using pre- or post-increment on non-integral types.
Did you change it because the atomic view tests actually have broader coverage for arithmetic types and you got a compilation error or because you realized it could not be instantiated for floating points?

One caveat with these changes is that it comes with a regression for pointer types.

Reply:
I just saw that we were missing overloads for `inc` and `dec` and didn't check test coverage.

Reply:
Then it probably means we never instantiate it.